server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -503,11 +502,11 @@ private boolean isReadyForUpload() {
             if (indexShard.getEngineOrNull() != null) {
                 sb.append(" engineType=").append(indexShard.getEngine().getClass().getSimpleName());
             }
-            if (isLocalOrSnapshotRecovery() == false) {
+            if (indexShard.recoveryState() != null) {
```

Comment:
isLocalOrSnapshotRecovery() checks for recoveryState. recoveryState can be null in which case this check throw NPE. Hence adding this check.

Reply:
You can always re-add `isLocalOrSnapshotRecovery` after a null check?

Reply:
yes, but we need to log the variables regardless of the final return value.